elide-async/src/main/java/com/yahoo/elide/async/operation/AsyncQueryOperation.java

```diff
@@ -21,4 +31,14 @@
      */
     public abstract Integer calculateRecordCount(AsyncQuery queryObj, ElideResponse response);
 
+    /**
+     * Check if Elide Response is NULL.
+     * @param response ElideResponse object.
+     * @throws NoHttpResponseException NoHttpResponseException Exception.
+     */
+    public void nullResponseCheck(ElideResponse response) throws NoHttpResponseException {
+        if (response == null) {
+            throw new NoHttpResponseException("Response for request returned as null");
```

Comment:
We usually don't throw library exceptions this deep in Elide.  What does the server generate here?

Reply:
Since this is just caught and then rethrown as IllegalStateException, is there any reason not to throw that directly here?    What do you expect the service to return to the end user in this scenario?